### ISSUE TITLE
valid_docs on convenience init method with unknown cause

### ISSUE THREAD

**Issue by Tableau-David-Potter** [realm/SwiftLint#705]
Using swiftlint 0.11.1, I'm getting a valid_docs warning with the following code:

``` swift
/// Returns an `NSDate` object initialized relative to 00:00:00 UTC on 1 January 1970 by a given number of
/// milliseconds.
///
/// - parameters:
///     - milliseconds:
///         The number of milliseconds from the reference date (00:00:00 UTC on 1 January 1970) for the new date.
///         Use a negative argument to specify a date and time before the reference date.
convenience init(millisecondsSince1970 milliseconds: Int64) {

    let doubleMilliseconds = Double(milliseconds)
    let seconds            = doubleMilliseconds / 1000
    self.init(timeIntervalSince1970: seconds)
}
```

If I replace the top description with the following, the problem goes away:

``` swift
/// Generates an `NSDate` object initialized relative to 00:00:00 UTC on 1 January 1970 by a given number of
/// milliseconds.
```

It looks like swiftlint has a problem with "Returns", but only in this case.  Here's a case where it works just fine (meaning I don't get the warning):

``` swift
/// Returns a `DataSetRow` from a row retrieved by calling `prepareRows()`.
///
/// - parameters:
///     - row:            A row from this table.
///     - dataSetColumns: Data set columns for creating the `DataSetRow`.
///
/// - returns: A `DataSetRow`.
func dataSetRowFromTableRow(row: Row, dataSetColumns: [DataSetColumn]) -> DataSetRow {
```

As a side note, it would be awesome if swiftlint would tell me why the code is failing the rule.

**Comment by Grubas7** [comment]
@Tableau-David-Potter There is known issue for returns in constructors: https://github.com/realm/SwiftLint/pull/739
Is this covering your problem also?

**Comment by Tableau-David-Potter** [comment]
Yes, that fixes it.  Thanks so much for the fix and for letting me know about it.

**Comment by marcelofabri** [comment]
Closing this since we removed this rule.
